tests/integration/ha_tests/application-charm/src/charm.py

```diff
@@ -0,0 +1,163 @@
+            self.on.clear_continuous_writes_action, self._on_clear_continuous_writes_action
+        )
+        self.framework.observe(
+            self.on.start_continuous_writes_action, self._on_start_continuous_writes_action
+        )
+        self.framework.observe(
+            self.on.stop_continuous_writes_action, self._on_stop_continuous_writes_action
+        )
+
+        # PID of the continuous writes OS process.
+        self._stored.set_default(continuous_writes_pid=None)
+
+    @property
+    def _connection_string(self) -> Optional[str]:
+        """Returns the PostgreSQL connection string."""
+        data = list(self.database.fetch_relation_data().values())[0]
+        username = data.get("username")
+        password = data.get("password")
+        endpoints = data.get("endpoints")
+        if None in [username, password, endpoints]:
+            return None
+
+        host = endpoints.split(":")[0]
+        return (
+            f"dbname='{self.database_name}' user='{username}'"
+            f" host='{host}' password='{password}' connect_timeout=5"
+        )
+
+    def _on_start(self, _) -> None:
+        """Only sets an Active status."""
+        self.unit.status = ActiveStatus()
+
+    def _on_database_created(self, _) -> None:
+        """Event triggered when a database was created for this application."""
+        self._start_continuous_writes(1)
+
+    def _on_endpoints_changed(self, _) -> None:
+        """Event triggered when the read/write endpoints of the database change."""
+        count = self._count_writes()
+        self._start_continuous_writes(count + 1)
+
+    def _count_writes(self) -> int:
+        """Count the number of records in the continuous_writes table."""
+        with psycopg2.connect(
+            self._connection_string
+        ) as connection, connection.cursor() as cursor:
+            cursor.execute("SELECT COUNT(number) FROM continuous_writes;")
+            count = cursor.fetchone()[0]
+        connection.close()
+        return count
+
+    def _on_clear_continuous_writes_action(self, _) -> None:
+        """Clears database writes."""
+        self._stop_continuous_writes()
+        with psycopg2.connect(
+            self._connection_string
+        ) as connection, connection.cursor() as cursor:
+            cursor.execute("DROP TABLE continuous_writes;")
+        connection.close()
+
+    def _on_start_continuous_writes_action(self, _) -> None:
+        """Start the continuous writes process."""
+        self._start_continuous_writes(1)
+
+    def _on_stop_continuous_writes_action(self, event: ActionEvent) -> None:
+        """Stops the continuous writes process."""
+        writes = self._stop_continuous_writes()
+        event.set_results({"writes": writes})
+
+    def _start_continuous_writes(self, starting_number: int) -> None:
+        """Starts continuous writes to PostgreSQL instance."""
+        if self._connection_string is None:
+            return
+
```

Comment:
Nit add '# stop any writes that might be going'

Reply:
Nice. Added on https://github.com/canonical/postgresql-operator/pull/36/commits/e9c2353a0537616df94c53f7466d86b307549de9.